src/traversals/greedy_color.jl

```diff
@@ -11,102 +11,273 @@ end
 best_color(c1::coloring, c2::coloring) = c1.num_colors < c2.num_colors ? c1 : c2
 
 """
-    perm_greedy_color(g, seq)
+   get_distinct_colors(g, v, seen, max_col) 
+
+Get number of distinct colors adjacent to `v`.
+Currently, the graph is colored by `max_col` vertices.
+`seen[i]` == true iff node `i` has been colored.
+"""
+function get_distinct_colors(
+    g::AbstractGraph,
+    v::T, cols::Vector{T},
+    seen::Vector{Bool},
+    max_col::T
+    ) where T <: Integer 
+
+    col_seen = zeros(Bool, max_col)
+    num_distinct = zero(T)
+    for i in neighbors(g, v)
+        if seen[i]
+            col_seen[cols[i]] || (num_distinct += one(T))
+            col_seen[cols[i]] = true
+        end
+    end
+    return num_distinct
+end
+
+"""
+    smallest_valid_color(g, v, max_col, cols, seen)
+
+Find the smallest color that none of the neighbors of `v` possess.
+Currently, the graph is colored by `max_col` vertices.
+`seen[i]` == true iff node `i` has been colored.
+"""
+function smallest_valid_color(
+    g::AbstractGraph,
+    v::T, max_col::T,
+    cols::Vector{T},
+    seen::Vector{Bool}
+    ) where T <: Integer 
+
+    to_consider = min(degree(g, v), max_col)
+    colors_used = zeros(Bool, to_consider)
+    for w in neighbors(g, v)
+        if seen[w] && cols[w] <= to_consider
+            colors_used[cols[w]] = true
+        end
+    end
+
+    best_col = to_consider+one(T)
+    for i in one(T):to_consider
+        if !colors_used[i]
+            best_col = i
+            break
+        end
+    end
+    return best_col
+end
+
+"""
+    invalidate_distinct_colors!(g, v, valid_distinct_cols)
+
+Set the `valid_distinct_cols` of the the neighbors of `v` to be invalid.
+Used after assigning `v` its color. 
+"""
+function invalidate_distinct_colors!(
+    g::AbstractGraph,
+    v::T, 
+    valid_distinct_cols::Vector{Bool}
+    ) where T <: Integer
+    
+    @inbounds @simd for u in neighbors(g, v)
+        valid_distinct_cols[u] = false    
+    end
```

Comment:
what happens if you replace this with
```
valid_distinct_cols[neighbors(g, v)] = false
```
?

Reply:
Improves run-time.